bindings/python/edlib.pyx

```diff
@@ -152,3 +154,91 @@ def align(query, target, mode="NW", task="distance", k=-1, additionalEqualities=
+           than 256 unique values
+    @param {str or bytes or iterable of hashable objects} target, combined with query must have no more
+           than 256 unique values
+    @param {string} gapSymbol, default "-"
+           String used to represent gaps in the alignment between query and target
+    @return Dictionary with following fields:
+            {string} query_aln  query_aln is the aligned query sequence from the input 'query'
+            {string} match_aln  match_aln is the pairwise alignment from the parsed cigar
+                The symbol '|' represents a read match '='
+                The symbol '.' represents a read mismtach 'X'
+                The space ' ' represents an insertion 'I' or deletion 'D'
+            {string} target_aln  target_aln is the aligned target sequence from the input 'target'                
+    """
+
+    ## 'path' - find edit distance, start and end locations and alignment path. (Default)
+    
+    ## access locations, a list of a set [(start, end)], from yech1990
+    ## target_pos = locations[0][0]
+    target_pos = alignResult["locations"][0][0]
+    if target_pos == None:
+        target_pos = 0
+    query_pos = 0 ## 0-indexed
+    target_aln = match_aln = query_aln = ""
+
+    cigar = alignResult["cigar"]
+
+    ###
+    ## cigar parsing, motivated by yech1990: https://github.com/Martinsos/edlib/issues/127
+    ##
+    ## 'num_occurences' == Number of occurrences of the alignment operation
+    ## 'alignment_operation' == Cigar symbol/code that represent an alignment operation
+    ## 
+    for num_occurrences, alignment_operation in re.findall("(\d+)(\D)", cigar):
+        num_occurrences = int(num_occurrences)
+        if alignment_operation == "=":
+            target_aln += target[target_pos : target_pos + num_occurrences]
+            target_pos += num_occurrences
+            query_aln += query[query_pos : query_pos + num_occurrences]
+            query_pos += num_occurrences
+            match_aln += "|" * num_occurrences
+        elif alignment_operation == "X":
+            target_aln += target[target_pos : target_pos + num_occurrences]
+            target_pos += num_occurrences
+            query_aln += query[query_pos : query_pos + num_occurrences]
+            query_pos += num_occurrences
+            match_aln += "." * num_occurrences
+        elif alignment_operation == "D":
+            target_aln += target[target_pos : target_pos + num_occurrences]
+            target_pos += num_occurrences
+            query_aln += gapSymbol * num_occurrences
+            query_pos += 0
+            match_aln +=gapSymbol * num_occurrences
+        elif alignment_operation == "I":
+            target_aln += gapSymbol * num_occurrences
+            target_pos += 0
+            query_aln += query[query_pos : query_pos + num_occurrences]
+            query_pos += num_occurrences
+            match_aln += gapSymbol * num_occurrences
+        else:
+            pass
```

Comment:
Can it be something else? This should never happen, right? If so, instead of passing, let's throw. Ohterwise we could hide errors, or produce result for faulty cigar and user will never know.

Reply:
I don't think it can be anything else, but it's possible there are odd cigar's out there in fields I don't know about. 

I'll throw an error here instead of passing. If there are issues, users should open a github PR.